File: day8/part1.py
from os import path

with open(path.join(path.dirname(__file__), 'input.txt'), 'r') as file:
    input = file.read().splitlines()

unique_segment_lens = [2, 3, 4, 7]
digits = [part for line in input for part in line.partition('|')[2].split() if len(part) in unique_segment_lens]

print(len(digits))

File: day10/part1.js
const fs = require('fs');
const path = require('path');

const input = fs.readFileSync(path.join(__dirname, 'input.txt')).toString().trim().split('\n');

const pairs = {
    '(': ')',
    '[': ']',
    '{': '}',
    '<': '>',
}

const scores = {
    ')': 3,
    ']': 57,
    '}': 1197,
    '>': 25137,
}

const errorScores = input.map(line => {
    const stack = [];

    for (const char of line) {
        if (Object.keys(pairs).includes(char)) {
            stack.push(char);
        } else {
            if (char !== pairs[stack.pop()]) {
                return scores[char];
            }
        }
    }

    return 0;
});

console.log(errorScores.reduce((a, b) => a + b, 0));
